main.go

```diff
@@ -99,7 +108,16 @@ func main() {
 		setupLog.Error(err, "unable to create controller", "controller", "ProgressiveSync")
 		os.Exit(1)
 	}
-	// +kubebuilder:scaffold:builder
+	//+kubebuilder:scaffold:builder
+
+	if err := mgr.AddHealthzCheck("healthz", healthz.Ping); err != nil {
+		setupLog.Error(err, "unable to set up health check")
+		os.Exit(1)
+	}
+	if err := mgr.AddReadyzCheck("readyz", healthz.Ping); err != nil {
```

Comment:
this is a very interesting addition.
Worth adding an issue to specialise the ready check?
For instance, making sure it can read the argo config etc

Reply:
Yes! :)